mesonbuild/compilers/compilers.py

```diff
@@ -915,7 +905,15 @@ def get_gcc_soname_args(gcc_type, prefix, shlib_name, suffix, path, soversion, i
         if soversion is not None:
             install_name += '.' + soversion
         install_name += '.dylib'
-        return ['-install_name', os.path.join('@rpath', install_name)]
+        args = ['-install_name', os.path.join(libdir, install_name)]
```

Comment:
Same as already noted above, as it is related to the rpath changes, please split this into a different commit.